src/app/services/building.service.ts

```diff
@@ -354,4 +354,42 @@ export class BuildingService {
       throw error;
     }
   }
+
+/**
+ * Adds a new operation to a building in the Firebase Realtime Database.
+ * If the building already has an 'operations' attribute, the function adds the operation to it.
+ * If the building doesn't have an 'operations' attribute, the function creates it and then adds the operation.
+ *
+ * @param buildingId - The ID of the building to which the operation will be added.
+ * @param operation - The operation object to add to the building.
+ * @returns A Promise that resolves when the operation is successfully added to the building.
+ * @throws Error if there is an issue adding the operation or the building is not found.
+ */
+  async addOperation(buildingId: string, operation: Operation): Promise<void> {
```

Comment:
Useless/Unnecessarily complex  function since the updateBuilding(building: Building) and the getBuilding(buildingId: string) function already exist

Reply:
same could be said for what is below (not related to this pr specifically)
![image](https://github.com/amczuboka/Decagon-CondoManagementSystem/assets/96556635/ee557b3d-f324-43e9-afa2-6927385fad15)

Reply:
I didn't want to update the whole building I wanted to do the modifications in 1 node to be more efficient thats  why i created this function

Reply:
I agree with Paul's reasoning here and I think the function looks good

Reply:
fully disagree with the reasoning because we do not generate enough traffic for it to be a problem. But the unit test work and whatnot so nvm